tests/jobs/ahm_analysis/test_ahm_analysis.py

```diff
@@ -0,0 +1,418 @@
+import os
+
+import pytest
+import pandas as pd
+import numpy as np
+
+import semeio.workflows.ahm_analysis.ahmanalysis as ahmanalysis
+from semeio._exceptions.exceptions import ValidationError
+
+from unittest.mock import MagicMock
+from itertools import product
+from pathlib import Path
+
+from scipy import stats
+from ecl.grid import EclGridGenerator
+
+
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_make_update_log_df(test_data_root):
+    """test function creates a dataframe from update_log file
+    and replace '...' with key_obs"""
+    test_data_dir = os.path.join(test_data_root, "update_log/oneobs")
+    update_log_path = test_data_dir
+    key_obs = "RWI_3_OBS"
+    updatelog_obs = ahmanalysis.make_update_log_df(update_log_path, key_obs)
+    assert "..." not in updatelog_obs["obs_key"]
+    assert updatelog_obs.at[0, "status"] == "Active"
+    assert updatelog_obs.at[1, "status"] == "Inactive"
+    assert (
+        updatelog_obs.columns
+        == [
+            "obs_number",
+            "obs_key",
+            "obs_mean",
+            "obs_std",
+            "status",
+            "sim_mean",
+            "sim_std",
+        ]
+    ).all()
+
+
+@pytest.mark.parametrize(
+    "input_dir, obs_key, expected_result",
+    [
+        ("update_log/oneobs", "RWI_3_OBS", "2 active/3"),
+        ("update_log/allobs", "ALL_OBS", "41 active/47"),
+        ("update_log/noactive", "RWI_2_OBS", "0 active/3"),
+    ],
+)
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_list_active_observations(test_data_root, input_dir, expected_result, obs_key):
```

Comment:
The `list_active_observations` takes a dataframe, I would consider avoiding to use `make_update_log_df` to create the input, and rather add the input data directly in `parametrize`. Would make it much easier to see what `list_active_observations` does as well, and why the expected results is as it is.

Reply:
I have updated the test.